erdos/src/node/leader.rs

```diff
@@ -221,6 +301,181 @@ impl Leader {
+                LeaderNotification,
+            >,
+        ) {
+            match msg_from_leader {
+                WorkerHandlerNotification::ScheduleJob(
+                    job_graph_id,
+                    job,
+                    worker_id,
+                    worker_addresses,
+                ) => {
+                    // The Leader assigns an operator to a worker.
+                    if id_of_this_worker == worker_id {
+                        tracing::trace!(
+                            "[WorkerHandler {}] Scheduling Job {:?} from JobGraph {:?}.",
+                            id_of_this_worker,
+                            job,
+                            job_graph_id,
+                        );
+                        if let Err(error) = worker_tx
+                            .send(LeaderNotification::ScheduleJob(
+                                job_graph_id,
+                                job,
+                                worker_addresses,
+                            ))
+                            .await
+                        {
+                            tracing::warn!(
+                                "[WorkerHandler {}] Error notifying Worker \
+                                        of the scheduled Job {:?}: {:?}",
+                                id_of_this_worker,
+                                job,
+                                error,
+                            );
+                        }
+                    }
+                }
+                WorkerHandlerNotification::ExecuteGraph(job_graph_id, worker_addresses) => {
+                    // Tell the Worker to execute the operators for this graph.
+                    if worker_addresses.contains(&id_of_this_worker) {
+                        tracing::debug!(
+                            "[WorkerHandler {}] Notifiying Worker to execute the JobGraph {:?}.",
+                            id_of_this_worker,
+                            job_graph_id,
+                        );
+                        if let Err(error) = worker_tx
+                            .send(LeaderNotification::ExecuteGraph(job_graph_id.clone()))
+                            .await
+                        {
+                            tracing::warn!(
+                                "[WorkerHandler {}] Error notifying Worker to \
+                                            execute JobGraph {:?}: {:?}",
+                                id_of_this_worker,
+                                job_graph_id,
+                                error,
+                            )
+                        }
+                    }
+                }
+                WorkerHandlerNotification::QueryResponse(worker_id, query_id, query_response) => {
+                    // The Leader responds to a Query by a Worker.
+                    if id_of_this_worker == worker_id {
+                        tracing::trace!(
```

Comment:
This was intentional since we expect queries to come in very often and should be mostly opaque in a debug log, whereas graphs are submitted pretty infrequently and it would be helpful to know when the message comes in for debugging.